neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -61,6 +62,37 @@ public UInt256 Hash
             Attributes.GetVarSize() +   //Attributes
             Witnesses.GetVarSize();     //Witnesses
 
+        public void CalculateGas()
+        {
+            if (Sender is null) Sender = UInt160.Zero;
+            if (Attributes is null) Attributes = new TransactionAttribute[0];
+            if (Witnesses is null) Witnesses = new Witness[0];
+            _hash = null;
+            long consumed;
+            using (ApplicationEngine engine = ApplicationEngine.Run(Script, this))
+            {
+                if (engine.State.HasFlag(VMState.FAULT))
+                    throw new InvalidOperationException();
+                consumed = engine.GasConsumed;
+            }
+            _hash = null;
+            long d = (long)NativeContract.GAS.Factor;
+            Gas = consumed - ApplicationEngine.GasFree;
+            if (Gas <= 0)
+            {
+                Gas = 0;
+            }
+            else
+            {
+                long remainder = Gas % d;
+                if (remainder == 0) return;
+                if (remainder > 0)
+                    Gas += d - remainder;
```

Comment:
The previous logic was:

```suggestion
                    Gas -= remainder + d;
```

Reply:
I don't think so...

The old is:

https://github.com/neo-project/neo/blob/8c7a12af75de5ff4816fdc23d92a35e9e0a77be8/neo/Network/P2P/Payloads/Transaction.cs#L108